src/mesh/voxel_cloud.rs

```diff
@@ -846,4 +831,161 @@ mod tests {
+
+        let voxel = voxel_cloud
+            .voxel_at_relative_coords(&Point3::new(0, 0, 0))
+            .unwrap();
+
+        assert!(!voxel);
+    }
+
+    #[test]
+    fn test_voxel_cloud_resize_zero_to_nonzero_correct_start_and_dimensions() {
+        let mut voxel_cloud = VoxelCloud::new(
+            &Point3::origin(),
+            &Vector3::zeros(),
+            &Vector3::new(1.0, 1.0, 1.0),
+        );
+        let new_origin = Point3::new(1, 2, 3);
+        let new_block_dimensions = Vector3::new(4, 5, 6);
+        voxel_cloud.resize(&new_origin, &new_block_dimensions);
+
+        assert_eq!(voxel_cloud.block_start(), new_origin);
+        assert_eq!(voxel_cloud.block_dimensions(), new_block_dimensions);
+        assert_eq!(voxel_cloud.voxel_map.len(), 4 * 5 * 6);
+    }
+
+    #[test]
+    fn test_voxel_cloud_resize_nonzero_to_zero_correct_start_and_dimensions() {
+        let mut voxel_cloud = VoxelCloud::new(
+            &Point3::new(1, 2, 3),
+            &Vector3::new(4, 5, 6),
+            &Vector3::new(1.0, 1.0, 1.0),
+        );
+        let new_origin = Point3::origin();
+        let new_block_dimensions = Vector3::zeros();
+        voxel_cloud.resize(&new_origin, &new_block_dimensions);
+
+        assert_eq!(voxel_cloud.block_start(), new_origin);
+        assert_eq!(voxel_cloud.block_dimensions(), new_block_dimensions);
+        assert_eq!(voxel_cloud.voxel_map.len(), 0);
+    }
+
+    #[test]
+    fn test_voxel_cloud_resize_nonzero_to_smaller_nonzero_correct_start_and_dimensions() {
+        let mut voxel_cloud = VoxelCloud::new(
+            &Point3::origin(),
+            &Vector3::new(4, 5, 6),
+            &Vector3::new(1.0, 1.0, 1.0),
+        );
+        let new_origin = Point3::new(1, 2, 3);
+        let new_block_dimensions = Vector3::new(1, 2, 3);
+        voxel_cloud.resize(&new_origin, &new_block_dimensions);
+
+        assert_eq!(voxel_cloud.block_start(), new_origin);
+        assert_eq!(voxel_cloud.block_dimensions(), new_block_dimensions);
+        assert_eq!(voxel_cloud.voxel_map.len(), 1 * 2 * 3);
+    }
+
+    #[test]
+    fn test_voxel_cloud_resize_nonzero_to_larger_nonzero_correct_start_and_dimensions() {
+        let mut voxel_cloud = VoxelCloud::new(
+            &Point3::origin(),
+            &Vector3::new(1, 2, 3),
+            &Vector3::new(1.0, 1.0, 1.0),
+        );
+        let new_origin = Point3::new(1, 2, 3);
+        let new_block_dimensions = Vector3::new(4, 5, 6);
+        voxel_cloud.resize(&new_origin, &new_block_dimensions);
+
+        assert_eq!(voxel_cloud.block_start(), new_origin);
+        assert_eq!(voxel_cloud.block_dimensions(), new_block_dimensions);
+        assert_eq!(voxel_cloud.voxel_map.len(), 4 * 5 * 6);
+    }
+
+    #[test]
+    fn test_voxel_cloud_resize_nonzero_to_larger_nonzero_grown_contains_false_rest_original() {
+        let original_origin = Point3::new(cast_i32(0), cast_i32(0), cast_i32(0));
```

Comment:
These `cast_i32` are almost certainly not needed. If rustc failed to infer the number type, you can either use type annotations or number literal suffixes like `0i32`.